src/Symfony/Component/Lock/Store/MysqlStore.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Doctrine\DBAL\Connection;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MysqlStore is a StoreInterface implementation using MySQL/MariaDB GET_LOCK function.
+ *
+ * @author Jérôme TAMARELLE <jerome@tamarelle.net>
+ */
+class MysqlStore implements StoreInterface
+{
+    /**
+     * @var \PDO|Connection
+     */
+    private $connection;
+    private $waitTimeout;
+
+    /**
+     * @param \PDO|Connection $connection
+     * @param int             $waitTimeout Time in seconds to wait for a lock to be released, for non-blocking lock.
```

Comment:
To be acquired, and not to be released right?

Reply:
It will be released after the timeout so that sounds right to me. You could also say "The time in seconds the lock is acquired for".